tutorials/dataframe/df106_HiggsToFourLeptons.C

```diff
@@ -0,0 +1,281 @@
+/// Systematic uncertainties for those MC scale factors are evaluated and the Vary function of RDataFrame is used to
+/// scale the variations to the final four leptons mass distribution.
+///
+///
+/// \macro_code
+/// \macro_image
+/// \date March 2020
+/// \author Stefan Wunsch (KIT, CERN)
+/// \date August 2022
+/// \author Julia Mathe (CERN)
+/// \date August 2023
+/// \author Marta Czurylo (CERN)
+
+#include <Math/Vector4D.h>
+#include <Math/Interpolator.h>
+#include <ROOT/RDFHelpers.hxx>
+#include <ROOT/RDataFrame.hxx>
+#include <ROOT/RVec.hxx>
+#include <TCanvas.h>
+#include <TH1D.h>
+#include <THStack.h>
+#include <TLatex.h>
+#include <TLegend.h>
+#include <TProfile.h>
+#include <TStyle.h>
+
+using namespace ROOT::VecOps;
+using ROOT::RVecF;
+using namespace ROOT::RDF::Experimental;
+
+// Define functions needed in the analysis
+// Select events for the analysis
+bool GoodElectronsAndMuons(const ROOT::RVecI &type, RVecF pt, RVecF eta, RVecF phi, RVecF e, RVecF trackd0pv,
+                           RVecF tracksigd0pv, RVecF z0)
+{
+   for (size_t i = 0; i < type.size(); i++) {
+      ROOT::Math::PtEtaPhiEVector p(0.001 * pt[i], eta[i], phi[i], 0.001 * e[i]);
+      if (type[i] == 11) {
+         if (pt[i] < 7000 || abs(eta[i]) > 2.47 || abs(trackd0pv[i] / tracksigd0pv[i]) > 5 ||
+             abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      } else {
+         if (abs(trackd0pv[i] / tracksigd0pv[i]) > 5 || abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      }
+   }
+   return true;
+}
+
+// Compute the invariant mass of a four-lepton-system.
+float ComputeInvariantMass(RVecF pt, RVecF eta, RVecF phi, RVecF e)
+{
+   ROOT::Math::PtEtaPhiEVector p1(pt[0], eta[0], phi[0], e[0]);
+   ROOT::Math::PtEtaPhiEVector p2(pt[1], eta[1], phi[1], e[1]);
+   ROOT::Math::PtEtaPhiEVector p3(pt[2], eta[2], phi[2], e[2]);
+   ROOT::Math::PtEtaPhiEVector p4(pt[3], eta[3], phi[3], e[3]);
+   return 0.001 * (p1 + p2 + p3 + p4).M();
+}
+
+void df106_HiggsToFourLeptons()
+{
+
+   // Enable Multithreading
+   ROOT::EnableImplicitMT();
+
+   // Create the RDataFrame from the spec json file. The df106_HiggsToFourLeptons_spec.json is provided in the same
+   // folder as this tutorial
+   std::string my_spec = gROOT->GetTutorialsDir() + std::string("/dataframe/df106_HiggsToFourLeptons_spec.json");
+   ROOT::RDataFrame df = ROOT::RDF::Experimental::FromSpec(my_spec);
+
+   // Add the ProgressBar feature
+   ROOT::RDF::Experimental::AddProgressbar(df);
+
+   // Perform the analysis
+   // Access metadata information that is stored in the JSON config file of the RDataFrame
+   // The metadata contained in the JSON file is accessible within a `DefinePerSample` call, through the `RDFSampleInfo`
+   // class
+   auto df_analysis =
+      df.DefinePerSample("xsecs", [](unsigned int slot, const ROOT::RDF::RSampleInfo &id) { return id.GetD("xsecs"); })
```

Comment:
```suggestion
   using ROOT::RDF::RSampleInfo;

   // Perform the analysis
   // Access metadata information that is stored in the JSON config file of the RDataFrame
   // The metadata contained in the JSON file is accessible within a `DefinePerSample` call, through the `RDFSampleInfo`
   // class
   auto df_analysis =
      df.DefinePerSample("xsecs", [](unsigned int slot, const RSampleInfo &id) { return id.GetD("xsecs"); })
```
Depending on your taste, this `RSampleInfo` is used so much here that it can be worth to avoid having to type out the namespace, to make the actual analysis code more brief.